### PR TITLE
Adding support for `array` key in raw dictionnaries in ASR pipeline.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

Adding support for `array` key in raw dictionnaries in ASR pipeline.
This means we can simplify the Quicktour example to look again a bit
more what the old `ffmpeg` example looked like.

Even simpler code might be enabled by
https://github.com/huggingface/datasets/issues/4180
in the future. We could then remove the `[:4]` which is currently necessary to prevent
loading a huge audio in RAM upfront.

Performance-wise the iterator also enables using stuff like `batch_size` and `num_workers` in the
pipeline object to get optimal performance (this will only show real differences on GPU which is not
included in this quicktour)

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@patrickvonplaten
@sgugger

### CODE REVIEW

File: docs/source/en/quicktour.mdx [hunk 1]
```diff
@@ -134,8 +134,7 @@ Audio files are automatically loaded and resampled when calling the `"audio"` co
 Let's extract the raw waveform arrays of the first 4 samples and pass it as a list to the pipeline:
 
 ```py
->>> raw_audio_waveforms = [d["array"] for d in dataset[:4]["audio"]]
->>> speech_recognizer(raw_audio_waveforms)
+>>> speech_recognizer(dataset[:4]["audio"])
```
Comment (patrickvonplaten) [inline docs/source/en/quicktour.mdx:137]: Nice! Could you update the `es` version as well?

**Review by patrickvonplaten — APPROVED** [review]

File: docs/source/en/quicktour.mdx (hunk 1 above)
Comment (Narsil) [reply docs/source/en/quicktour.mdx:137]: Done !

File: src/transformers/pipelines/automatic_speech_recognition.py [hunk 2]
```diff
@@ -193,7 +193,14 @@ def preprocess(self, inputs, chunk_length_s=0, stride_length_s=None):
         extra = {}
         if isinstance(inputs, dict):
             stride = inputs.pop("stride", None)
-            _inputs = inputs.pop("raw")
+            # Accepting `"array"` which is the key defined in `datasets` for
+            # better integration
+            if not ("sampling_rate" in inputs and ("raw" in inputs or "array" in inputs)):
+                raise ValueError(
+                    'When passing a dictionnary to AutomaticSpeechRecognitionPipeline, the dict needs to contain a "raw" key containing the numpy array representing the audio and a "sampling_rate" key, containing the sampling_rate associated with that array'
```
Comment (sgugger) [inline src/transformers/pipelines/automatic_speech_recognition.py:200]:
```suggestion
                    'When passing a dictionnary to AutomaticSpeechRecognitionPipeline, the dict needs to contain a '
                    '"raw" key containing the numpy array representing the audio and a "sampling_rate" key, '
                    "containing the sampling_rate associated with that array"
```
Let's split this on several lines to respect our 119 char limit :-)

**Review by sgugger — APPROVED** [review]
Thanks for fixing this!

**Pull request merged**